pkg/openstack/loadbalancer_test.go

```diff
@@ -973,3 +976,281 @@ func TestLbaasV2_updateServiceAnnotations(t *testing.T) {
+							"loadbalancer.openstack.org/class":     "test-class",
+						},
+					},
+				},
+			},
+			want: "annotation-test-id",
+		},
+		{
+			name: "test get subnet from config class",
+			args: args{
+				svcConf: &serviceConfig{},
+				lbaasV2: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBClasses: map[string]*LBClass{
+								"test-class": {
+									SubnetID: "test-class-subnet-id",
+								},
+							},
+						},
+					},
+				},
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							"loadbalancer.openstack.org/class": "test-class",
+						},
+					},
+				},
+			},
+			want: "test-class-subnet-id",
+		},
+		{
+			name: "test get subnet from config class with invalid loadbalancer class",
+			args: args{
+				svcConf: &serviceConfig{},
+				lbaasV2: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBClasses: map[string]*LBClass{
+								"decoy-class": {
+									SubnetID: "test-id",
+								},
+							},
+							SubnetID: "test-subnet-id",
+						},
+					},
+				},
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							"loadbalancer.openstack.org/class": "test-class",
+						},
+					},
+				},
+			},
+			want:        "",
+			expectedErr: fmt.Sprintf("invalid loadbalancer class %q", "test-class"),
+		},
+		{
+			name: "test get subnet from default config",
+			args: args{
+				svcConf: &serviceConfig{},
+				lbaasV2: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBClasses: map[string]*LBClass{
+								"test-config-class-subnet-id": {
+									SubnetID: "test-id",
+								},
+							},
+							SubnetID: "test-default-subnet-id",
+						},
+					},
+				},
+				service: &corev1.Service{},
+			},
+			want: "test-default-subnet-id",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := tt.args.lbaasV2.getSubnetID(tt.args.service, tt.args.svcConf)
+			if tt.expectedErr != "" || err != nil {
+				assert.EqualError(t, err, tt.expectedErr)
+			}
+			if tt.expectedErr == "" {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, got, tt.want)
+		})
+	}
+}
+
+func Test_buildMonitorCreateOpts(t *testing.T) {
+	type testArg struct {
+		lbaas   *LbaasV2
+		svcConf *serviceConfig
+		port    corev1.ServicePort
+	}
+	tests := []struct {
+		name    string
+		testArg testArg
+		want    v2monitors.CreateOpts
+	}{
+		{
+			name: "test for port protocol udp with ovn provider",
+			testArg: testArg{
+				lbaas: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBProvider: "ovn",
+						},
+						lb: &gophercloud.ServiceClient{
+							Endpoint: "/endpoint/",
+						},
+					},
+				},
+				svcConf: &serviceConfig{
+					healthMonitorDelay:          1,
+					healthMonitorTimeout:        1,
+					healthMonitorMaxRetries:     1,
+					healthMonitorMaxRetriesDown: 1,
```

Comment:
It would be better to have these values different to also check if function doesn't mix them.